# Epigenetics

The genetic material is known to be modified during the life of an organism, possibly causing modifications in gene behavior. Far is known about mutations being a mayor player in genetic variation and the profiling of these variations has proved to be highly useful when studying diseases and evolutionary theory [GWAS]. In eukaryotes, there are in addition mechanisms in which the DNA can be modified without altering the molecular sequence, so called epigenetic mechanisms. As Conrad Waddington, who coined the term "epigenetics", defined: "it is the branch of biology which studies the causal interactions between genes and their products, which bring the phenotype into being" [Waddington]. Such definition led to categorizing as epigenetics all biological phenomena which correlated the genetic material with the genetic products and were not explained entirely by the classic genetic studies. Further studies have revealed that epigenetic mechanisms can be modulated in response to external stimuli [CITATION], entailing an overlay between DNA  and environment for the cells and organisms. Moreover, the epigenetic mechanisms behavior can vary for different stages of cell development [CITATION], environmental changes [CITATION] or disease [CITATIONS]. In the same way genetic variability can be profiled, it is possible to decipher shared patterns for the epigenetic modifications in different scenarios and types of tissue. 

Due to the latest growth on research efforts and resources about the topic, we were able to characterized the inheritance of gene expression patterns not explained by the encoded information in the DNA sequence but through epigenetic modifications. High-throughput technologies such as Chromatin Inmunoprecipitation next-generation sequencing (ChIP-seq) or Whole-Genome Bisulfite sequencing (WGBS), allowed us to obtain an incredibly vast amount of information on epigenetic marks throughout the genome. It was then possible to determine the epigenome profile consisting of multiple chromatin states that activate or repress the gene expression in a local and cell-specific manner. This ``epigenomic profiling'' helped the understanding of cell differentiation, where even though the vast majority of cells in a multicellular organism share an identical genotype, the development of the various tissues generates stable but diverse profiles of gene expression, giving rise to the multiple cell types and differentiated cellular functions. In light of this, more specifically epigenetics may be defined as “the study of any potentially stable and, ideally, heritable change in gene expression or cellular phenotype that occurs without changes in Watson-Crick base-pairing of DNA” [CITATION Gol07].

The work on nucleic acids [ ], chromatin [ ] and histone proteins [ ] led to the understanding of the DNA arrangement, as being wrapped around the histone proteins [ ], and furthermore to the cytological distinction between euchromatin and heterochromatin [ ]. It has been proved that post-translational modifications of the histones, such as methylation, acetylation or phosphorylation, can reshape the chromatin structural and functional properties instating the concept of turning ``on'' and ``off'' regions of the genetic material [ ]. A challenging yet feasible task is to characterize those configurations responsible for the repression, activation or modulation of the gene expression via epigenetic modifications, both in different tissues and also when affected by diseases as in the case of the cancer cells used in this study. For further understanding, it is essential to know about the diverse ways epigenetic mechanisms work and elucidate the correlation among them and to biological processes.  

## Epigenetic modification types:

### Histone Modifications

Histones are proteins that are the primary components of chromatin, which is the complex of DNA and proteins that makes up chromosomes. Histones are arranged as a spool around which DNA can wind and these proteins can be modified chemically. Histone modifications first work, in particular histone acetylation [ CITATION All64 \l 3082 ], suggested a close relationship between histone modification state and the local gene activity, hypothesis that was subsequently supported by studies on histone-tail mutations in *Saccharomyces cerevisiae* [CITATION Kay \l 3082 ]*,* hypoacetulation of the inactive X chromosome in female mammals [ CITATION Jep93 \l 3082 ]*,* as well as hyperacetylation of the twofold upregulated X chromosome in *D. melanogaster* males [ CITATION Bon94 \l 3082 ]. These major findings led to the compelling argument that histone modifications, along with DNA methylation, contribute to distinguish between euchromatin state and heterochromatin. Accordingly, depending on the histone modification state, the chromatin can be arranged as euchromatin, so there is gene activity, or as heterochromatin, so there is gene repression.  

Chromatin state at promoters is largely invariant across diverse cell types, whereas enhancers are marked with highly cell-type-specific histone modification patterns [ CITATION Hei09 \l 3082 ]. As in methylation patterns, an aberrant histone modification pattern is associated with the development of cancer [ CITATION Mar01 \l 3082 ]. Histone deacetylases (HDACs) are implicated expecting both positive and negative effects on oncogenic and oncosuppressive mechanisms. Again, the importance of the histone modification patterns for the gene expression and the diseases associated with an aberrant histone modification profile, call our attention on the topic.

### DNA methylation

DNA methylation is perhaps the best characterized chemical modification of chromatin and were detected as early as 1984[ CITATION Hot48 \l 3082 ]. In mammals, nearly all DNA methylation occurs on cytosine residues of CpG dinucleotides. Regions of the genome that have a high density of CpGs are referred to as CpG islands, and DNA methylation of these islands correlates with transcriptional expression [ CITATION Raz80 \l 3082  \m Bir85]. De novo or maintenance DNA methyltransferases (DNMTs) play a critical role in gene regulation, especially those associated with transposons and imprinted genes [CITATION Gol05 \l 3082 ], by keeping the genomic patterns of cytosine methylation during embryogenesis and gametogenesis. Moreover, the formation of heterochromatin in many organisms is mediated in part by DNA methylation and its binding proteins in combination with RNA and histone modifications. DNA methylation takes part in many cellular processes including silencing of repetitive and centromeric sequences from fungi to mammals [CITATION Par02 \l 3082 ]; X chromosome inactivation in female mammals [ CITATION Lyo61 \l 3082 ]; and mammalian imprinting [ CITATION Sur84 \l 3082  \m McG84], all of which can be stably maintained. Taken together, DNA methylation provides a stable, heritable, and critical component of epigenetic regulation.  

As the other epigenetic mechanisms, DNA methylation is reversible and therefore DNA methylation patterns vary in time and space during differentiation [ CITATION Bir02 \l 3082 ]. However, abnormal control of the methylation pattern was detected in cancer cells and may result in the generation of random modification patterns which may serve to unleash new genes for transcription [ CITATION Jon86 \l 3082 ]. The abnormal methylation pattern can be either hypomehylated, which usually involves repeated DNA sequences, or hypermethylated which involves CpG islands [ CITATION Ehr02 \l 3082 ]. In the first case, oncogenes are activated whereas hypermethylation repress the transcription of the promoter regions of tumor suppressor genes, leading to gene silencing. Therefore, DNA methylation and cancer correlation makes this epigenetic mechanism very significant for a greater comprehension of gene expression regulation and diseases related to it

### RNA-Associated Silencing

RNA silencing is another method to turn off genes when it is in form of antisense transcripts, noncoding RNAs or RNA interference. Antisense double-stranded RNA complementary to targeted mRNAs was detected as a method of Post-Transcriptional gene silencing (PTGS) for both cellular and viral genes in a sequence-specific manner [ CITATION Ham99 \l 3082 ]. This kind of process is known as RNA-mediated interference or RNAi. Moreover, small non-coding RNAs were identified as potencial ‘templating’ molecules for the location-specific epigenetic modifications. Several researches reported the involvement of small nuclear RNAs (snRNAs) in interacting with and presumably directing chromatin-modifying activities [CITATION Vol02 \m Moc02 \m Mar \l 3082 ]. The snRNAs participate in a nuclear process known as ‘transcriptional gene silencing’ (TGS) guiding the epigenetic machinery not only for heterochromatin assembly and gene silencing [ CITATION Mar \l 3082 ] but also directing programmed DNA elimination [ CITATION Cha13 \l 3082 ].

### Alternative Splicing

Alternative splicing is on major mechanism that makes the most of the precursor messenger RNAs (pre-mRNAs) by processing the pre-mRNA into a diverse array of mature mRNAs that encode distinct proteins. This phenomenon explains the high complexity of organisms as humans while they have a relatively small number of protein-coding genes. Alternative splicing of RNA leads to a variety of possible mRNA isoforms and proteins, which can have different, and often opposing, functions (Figure 1). Sequences called exons are regions of the pre-mRNA that are included in the mature mRNA, such as the protein-coding sequences and regulatory untranslated regions at either end of the mRNA. Sequences called introns are the portions of the pre-mRNA that are removed during splicing. In alternative splicing, some sequences serve as exons under some conditions and are included in the final mRNA. At other times, however, the alternative splicing process may exclude the same sequence, treating it as an intron and removing it from the mature mRNA.

A critical finding regarding the prevalence of alternative splicing was that a majority of human genes produce a wide variety of messenger RNAs (mRNA) that in turn encode distinct proteins [ CITATION Joh03 \l 3082 ]. Scientists estimate that 15–60 percent of human genetic diseases involve splicing mutations, either through direct mutation of the splice-site signals or through disruption of other components of the splicing pathway [ CITATION Wan07 \l 3082 ]. Therefore, understanding how the splicing machinery distinguish between exons, which are part of the mature mRNA, and introns, which are removed from the pre-mRNA, is of critical importance. Alternative splicing adds an extra layer of complexity, because regulatory sequences that sometimes designate an exon's inclusion into the mature mRNA dictate the exclusion of that exon under other conditions.

\subsection{Epigenetic modelling}

