Errores (6/33):

 Cell line K562, Mark POLR2A.

 Cell line A549, Mark H3K4me3

 Cell line A549, Mark H3K4me3

Cell line A549, Mark H3K36me3

Cell line A549, Mark H3K9me3

Cell line A549, Mark H3K27me3



(mscthesis) [alekssro@fe1 mscthesis]$ ./scripts/epigeNMF.sh -c K562 A549 Hela-S3
Calculating V matrix of normalized counts from data in 'data/' directory 
Saving results in 'results/' directory

Counting reads:
  cell line -> K562     signal track -> H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K4me3
  cell line -> K562     signal track -> POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: POLR2A
  cell line -> K562     signal track -> H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K4me1
  cell line -> K562     signal track -> CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: CTCF
  cell line -> K562     signal track -> EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: EP300
  cell line -> K562     signal track -> H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K27ac
  cell line -> K562     signal track -> H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K36me3
  cell line -> K562     signal track -> H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K9me3
  cell line -> K562     signal track -> H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K27me3
  cell line -> K562     signal track -> H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H2A.Z
  cell line -> K562     signal track -> H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in K562: H3K9ac
  cell line -> A549     signal track -> H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K4me3
  cell line -> A549     signal track -> POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: POLR2A
  cell line -> A549     signal track -> H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K4me1
  cell line -> A549     signal track -> CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: CTCF
  cell line -> A549     signal track -> EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: EP300
  cell line -> A549     signal track -> H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K27ac
  cell line -> A549     signal track -> H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K36me3
  cell line -> A549     signal track -> H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K9me3
  cell line -> A549     signal track -> H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K27me3
  cell line -> A549     signal track -> H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H2A.Z
  cell line -> A549     signal track -> H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in A549: H3K9ac
  cell line -> Hela-S3  signal track -> H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K4me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K4me3
  cell line -> Hela-S3  signal track -> POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: POLR2A
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: POLR2A
  cell line -> Hela-S3  signal track -> H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K4me1
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K4me1
  cell line -> Hela-S3  signal track -> CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: CTCF
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: CTCF
  cell line -> Hela-S3  signal track -> EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: EP300
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: EP300
  cell line -> Hela-S3  signal track -> H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K27ac
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K27ac
  cell line -> Hela-S3  signal track -> H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K36me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K36me3
  cell line -> Hela-S3  signal track -> H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K9me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K9me3
  cell line -> Hela-S3  signal track -> H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K27me3
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K27me3
  cell line -> Hela-S3  signal track -> H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H2A.Z
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H2A.Z
  cell line -> Hela-S3  signal track -> H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K9ac
      Increase the total count of informative reads across all ChIP-seq assay samples in Hela-S3: H3K9ac


Calculating counts per genomic bin for Cell line K562, Mark H3K4me3.
  Intersect genomic bins with processed bed ENCFF681JQI.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF804NBY.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF681JQI.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF681JQI.rawCounts.bed  calculated
  Loading ENCFF804NBY.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF804NBY.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K4me3  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark POLR2A.
  Intersect genomic bins with processed bed ENCFF138LNW.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF839GLQ.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119:  5641 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  POLR2A 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF138LNW.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF138LNW.rawCounts.bed  calculated
  Loading ENCFF839GLQ.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line K562, Mark H3K4me1.
  Intersect genomic bins with processed bed ENCFF213ZYR.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF798WDB.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me1 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF213ZYR.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF213ZYR.rawCounts.bed  calculated
  Loading ENCFF798WDB.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF798WDB.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K4me1  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark CTCF.
  Intersect genomic bins with processed bed ENCFF487UYG.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF496SZR.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  CTCF 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF487UYG.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF487UYG.rawCounts.bed  calculated
  Loading ENCFF496SZR.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF496SZR.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark CTCF  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark EP300.
  Intersect genomic bins with processed bed ENCFF982AFE.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF200PYZ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  EP300 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF982AFE.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF982AFE.rawCounts.bed  calculated
  Loading ENCFF200PYZ.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF200PYZ.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark EP300  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H3K27ac.
  Intersect genomic bins with processed bed ENCFF384ZZM.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF070PWH.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF384ZZM.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF384ZZM.rawCounts.bed  calculated
  Loading ENCFF070PWH.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF070PWH.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K27ac  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H3K36me3.
  Intersect genomic bins with processed bed ENCFF190AQC.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF303VNU.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K36me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF190AQC.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF190AQC.rawCounts.bed  calculated
  Loading ENCFF303VNU.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF303VNU.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K36me3  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H3K9me3.
  Intersect genomic bins with processed bed ENCFF805FLY.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF038JTR.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF805FLY.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF805FLY.rawCounts.bed  calculated
  Loading ENCFF038JTR.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF038JTR.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K9me3  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H3K27me3.
  Intersect genomic bins with processed bed ENCFF937NEW.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF666LFR.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF937NEW.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF937NEW.rawCounts.bed  calculated
  Loading ENCFF666LFR.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF666LFR.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K27me3  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H2A.Z.
  Intersect genomic bins with processed bed ENCFF840EXN.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF289NOY.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H2A.Z 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF840EXN.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF840EXN.rawCounts.bed  calculated
  Loading ENCFF289NOY.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF289NOY.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H2A.Z  to the V matrix file 


Calculating counts per genomic bin for Cell line K562, Mark H3K9ac.
  Intersect genomic bins with processed bed ENCFF501UKW.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF355NBA.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF501UKW.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF501UKW.rawCounts.bed  calculated
  Loading ENCFF355NBA.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF355NBA.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K9ac  to the V matrix file 

Loading required package: data.table
Transposing V matrix...

Calculating counts per genomic bin for Cell line A549, Mark H3K4me3.
  Intersect genomic bins with processed bed ENCFF079TOP.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119: 47736 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
  Intersect genomic bins with processed bed ENCFF761OKC.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF079TOP.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line A549, Mark POLR2A.
  Intersect genomic bins with processed bed ENCFF239AAI.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF819TOT.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  POLR2A 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF239AAI.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF239AAI.rawCounts.bed  calculated
  Loading ENCFF819TOT.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF819TOT.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark POLR2A  to the V matrix file 


Calculating counts per genomic bin for Cell line A549, Mark H3K4me1.
  Intersect genomic bins with processed bed ENCFF336KSW.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119: 51842 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
  Intersect genomic bins with processed bed ENCFF921GLJ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me1 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF336KSW.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line A549, Mark CTCF.
  Intersect genomic bins with processed bed ENCFF528AJY.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF319XNL.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  CTCF 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF528AJY.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF528AJY.rawCounts.bed  calculated
  Loading ENCFF319XNL.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF319XNL.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark CTCF  to the V matrix file 


Calculating counts per genomic bin for Cell line A549, Mark EP300.
  Intersect genomic bins with processed bed ENCFF540LLB.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF864VSA.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  EP300 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF540LLB.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF540LLB.rawCounts.bed  calculated
  Loading ENCFF864VSA.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF864VSA.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark EP300  to the V matrix file 


Calculating counts per genomic bin for Cell line A549, Mark H3K27ac.
  Intersect genomic bins with processed bed ENCFF122XDP.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF265SHJ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF122XDP.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF122XDP.rawCounts.bed  calculated
  Loading ENCFF265SHJ.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF265SHJ.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K27ac  to the V matrix file 


Calculating counts per genomic bin for Cell line A549, Mark H3K36me3.
  Intersect genomic bins with processed bed ENCFF885LBR.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119:  3012 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
  Intersect genomic bins with processed bed ENCFF992DBZ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K36me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF885LBR.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line A549, Mark H3K9me3.
  Intersect genomic bins with processed bed ENCFF441ELG.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119:  4348 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
  Intersect genomic bins with processed bed ENCFF852IQZ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF441ELG.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line A549, Mark H3K27me3.
  Intersect genomic bins with processed bed ENCFF359QIW.processed.bed to get rawCounts
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
./scripts/bedToNormCounts.sh: line 119:  6329 Aborted                 (core dumped) bedtools intersect -a ${BINNED_GENOME} -b ${PROCESSED_FILEPATH} -c > ${OUT_DIR}/${SAMPLE_ID}.rawCounts.bed
  Intersect genomic bins with processed bed ENCFF060OKK.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF359QIW.rawCounts.bed for estimating normalized signal... 
Error in read.table(file = filepath, header = FALSE, sep = "\t", stringsAsFactors = FALSE) : 
  no lines available in input
Execution halted

Calculating counts per genomic bin for Cell line A549, Mark H2A.Z.
  Intersect genomic bins with processed bed ENCFF274LSQ.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF853MFU.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H2A.Z 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF274LSQ.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF274LSQ.rawCounts.bed  calculated
  Loading ENCFF853MFU.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF853MFU.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H2A.Z  to the V matrix file 


Calculating counts per genomic bin for Cell line A549, Mark H3K9ac.
  Intersect genomic bins with processed bed ENCFF017SPM.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF965KEO.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF017SPM.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF017SPM.rawCounts.bed  calculated
  Loading ENCFF965KEO.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF965KEO.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K9ac  to the V matrix file 

Loading required package: data.table
Transposing V matrix...

Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K4me3.
  Intersect genomic bins with processed bed ENCFF440OOW.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF475KTC.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF440OOW.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF440OOW.rawCounts.bed  calculated
  Loading ENCFF475KTC.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF475KTC.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K4me3  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark POLR2A.
  Intersect genomic bins with processed bed ENCFF452GJT.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF713XOH.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  POLR2A 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF452GJT.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF452GJT.rawCounts.bed  calculated
  Loading ENCFF713XOH.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF713XOH.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark POLR2A  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K4me1.
  Intersect genomic bins with processed bed ENCFF945WUY.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF395EGN.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K4me1 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF945WUY.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF945WUY.rawCounts.bed  calculated
  Loading ENCFF395EGN.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF395EGN.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K4me1  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark CTCF.
  Intersect genomic bins with processed bed ENCFF474UIC.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF163WNN.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  CTCF 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF474UIC.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF474UIC.rawCounts.bed  calculated
  Loading ENCFF163WNN.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF163WNN.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark CTCF  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark EP300.
  Intersect genomic bins with processed bed ENCFF092PCB.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF952XCP.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  EP300 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF092PCB.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF092PCB.rawCounts.bed  calculated
  Loading ENCFF952XCP.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF952XCP.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark EP300  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K27ac.
  Intersect genomic bins with processed bed ENCFF113QJM.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF218BUW.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF113QJM.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF113QJM.rawCounts.bed  calculated
  Loading ENCFF218BUW.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF218BUW.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K27ac  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K36me3.
  Intersect genomic bins with processed bed ENCFF562XBI.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF491XKR.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K36me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF562XBI.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF562XBI.rawCounts.bed  calculated
  Loading ENCFF491XKR.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF491XKR.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K36me3  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K9me3.
  Intersect genomic bins with processed bed ENCFF638EFN.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF380OGG.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF638EFN.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF638EFN.rawCounts.bed  calculated
  Loading ENCFF380OGG.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF380OGG.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K9me3  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K27me3.
  Intersect genomic bins with processed bed ENCFF576BBK.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF833PJA.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K27me3 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF576BBK.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF576BBK.rawCounts.bed  calculated
  Loading ENCFF833PJA.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF833PJA.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K27me3  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H2A.Z.
  Intersect genomic bins with processed bed ENCFF185PMC.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF645GVZ.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H2A.Z 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF185PMC.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF185PMC.rawCounts.bed  calculated
  Loading ENCFF645GVZ.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF645GVZ.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H2A.Z  to the V matrix file 


Calculating counts per genomic bin for Cell line Hela-S3, Mark H3K9ac.
  Intersect genomic bins with processed bed ENCFF203MBP.processed.bed to get rawCounts
  Intersect genomic bins with processed bed ENCFF112SRH.processed.bed to get rawCounts
Initiating normalization of counts for each bin to add a column in the V matrix
Read 2 items
Epignetic mark:  H3K9ac 
Read 2 items
  Loading uniqueness mappability track and calculating mappable genome size... 
  Loading 200-bp binned genome file... 
  Loading ENCFF203MBP.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF203MBP.rawCounts.bed  calculated
  Loading ENCFF112SRH.rawCounts.bed for estimating normalized signal... 
    Estimate expected counts by finding overlaps between bins and mappability regions 
    Normalized counts for  ENCFF112SRH.rawCounts.bed  calculated
  Appending combined counts for the epigenetic mark H3K9ac  to the V matrix file 

Loading required package: data.table
Transposing V matrix...