% Chapter Template

\chapter{Methods}\label{Methods} % for referencing this chapter elsewhere, use \ref{Methods}

%----------------------------------------------------------------------------------------
%	SECTION 1
%----------------------------------------------------------------------------------------

\section{NMF description}

Explain the statistical framework in detail

%-----------------------------------
%	SUBSECTION 1
%-----------------------------------
\subsection{Algorithms}

Explain the different types of algorithms to solve the problem and the one used.

%-----------------------------------
%	SUBSECTION 2
%-----------------------------------

\subsection{Choosing number of signatures}

Explain different ways to choose N, the way used, the N used and why.

%----------------------------------------------------------------------------------------
%	SECTION 2
%----------------------------------------------------------------------------------------

\section{Bio Justification}

[Change title]

Explain the relation between signatures found in NMF and biological processes

%----------------------------------------------------------------------------------------
%	SECTION 3
%----------------------------------------------------------------------------------------

\section{Data}

The data used in this study is obtained entirely from the ENCODE project database [ ], where multiple epigenetic marks and reference epigenomes are available. Data sets from 11 types of epigenetic marks were collected, including histone modifications (H3K4me3, H3K4me1, H3K27ac, H3K36me3, H3K9me3, H3K27me3 and H3K9ac), chromatin remodeling proteins (EP300 and H2A.Z) and transcription regulation factors (CTCF, POLR2A). Replicate samples for each of the 11 epigenetic marks were used in the three cell lines of study, which includes a human liver cancer cell line (HepG2) [ ], myelogenous leukemia cell line (K562) [ ] and cells derived from HeLa cancerous cervical tumor line (HeLa-S3).

\medskip

In order to standardize the input and facilitate the data sets processing, the information for all the samples was arranged in a tab separated file with the following fields as columns:

\begin{enumerate}
    \item \textbf{Cell line type}. Either HepG2, K562 or Hela-S3.
    \item \textbf{Epigenetic modification category}. Where does the modification apply. Either histone modification (`histonemod'), chromatin modulation (`openchromatin') or transcription factor (`TFBS').
    \item \textbf{Epigenetic modification name}. The label of the mark which the sample corresponds to, as mentioned above.
    \item \textbf{Accession ID}. The accession name for the sample in the database.
    \item \textbf{File name}.
    \item \textbf{Processing status}. Either `raw' or `process'. Raw samples need to be pre-processed.
    \item \textbf{Replicate number}. Integer enumerating the various replicates for a particular epigenetic mark.
    \item \textbf{Database name}. In the present case, `ENCODE'.
    \item \textbf{Download link}. Used to download the sample reads file.
\end{enumerate}

As a matter of convenience, `\path{downloadData.sh} \path{[DataInfo.tsv]}' or `\path{epigeNMF.sh} \path{-d}  \path{[DataInfo.tsv]}' can be used to automatically download the data sets into the appropriate directory tree: \path{CELL_LINE/SIGNAL_TRACK/SAMPLE_ID}.

%----------------------------------------------------------------------------------------
%	SECTION 4
%----------------------------------------------------------------------------------------

\section{Pipeline}

Explain the command line tool, the pipeline followed for the analysis and explain functioning of each script

You should follow the next pipeline for the analysis:

\begin{enumerate}
    \item Download the data (option \path{-d} or \path{--download})
    \item Prepare BED alignment files (option \path{-p} or \path{--process})
    \item Create V matrix of counts for marks (cols) by bins (rows) (option \path{-c} or \path{--counts})
    \item Filter V matrix bins to remove noise (option \path{-f} or \path{--filter})
    \item Choose the optimal `n' number of signatures for NMF (option \path{-k} or \path{--chooseN})
    \item NMF analysis (option \path{-n} or \path{--nmf})
\end{enumerate}

\section{Performance}

Perform analysis with percentage of the data and get performance (time) plots
